### PR TITLE
Accept IpamOptions via NetworkOptionIpam

### PR DESCRIPTION
**Pull request by rmb938**
Accept IpamOptions via NetworkOptionIpam. 

This change also removes subnet specific IPAM Options and moves them to per network. 

See pull docker/docker#17316 for corresponding changes and discussion

### CODE REVIEW

**Comment by rmb938** [comment]
Any idea why CircleCI is erroring? I can't seem to replicate the error and it isn't very descriptive so I have no idea whats wrong.

**Comment by aboch** [comment]
@rmb938 
Rebase to latest master. A change that went in yesterday fixed the issue you are hitting in circle ci.

**Comment by rmb938** [comment]
@aboch Thanks but that doesn't seem to fix it. Same error

**Comment by aboch** [comment]
Oh, the failure is a code style check failure:

```
Checking format... 
test -z "$(goimports -l . | grep -v Godeps/_workspace/src/ | tee /dev/stderr)"
libnetwork_internal_test.go
Makefile:55: recipe for target 'check-format' failed
make: *** [check-format] Error 1
```

**Comment by aboch** [comment]
Also please squash the two commits in one. They are not about separate things.

File: libnetwork_internal_test.go [hunk 1]
```diff
@@ -67,20 +67,19 @@ func TestNetworkMarshalling(t *testing.T) {
...
+			netlabel.MacAddress: "a:b:c:d:e:f",
+		},
 		ipamV4Config: []*IpamConf{
 			&IpamConf{
 				PreferredPool: "10.2.0.0/16",
 				SubPool:       "10.2.0.0/24",
-				Options: map[string]string{
-					netlabel.MacAddress: "a:b:c:d:e:f",
-				},
 				Gateway:      "",
```
Comment (aboch) [inline libnetwork_internal_test.go]: It's an uneven indentation here which causes the ci to fail.
If you run `goimports -w *.go` it will fix it.

**Comment by rmb938** [comment]
@aboch Well I can't seem to replicate the issue on my end. On my machine it checks successfully

```
Checking format...
test -z "$(goimports -l . | grep -v Godeps/_workspace/src/ | tee /dev/stderr)"
Done checking format
```

I don't understand what could be wrong formatting wise as goimports and gofmt seem happy.

File: libnetwork_internal_test.go (hunk 1 above)
Comment (rmb938) [inline libnetwork_internal_test.go]: Ah I see. Thanks

File: test/integration/dnet/multi.bats [hunk 2]
```diff
@@ -22,6 +22,7 @@ function is_network_exist() {
 	run dnet_cmd $(inst_id2port $i) network create -d test ${oname}
 	echo ${output}
 	[ "$status" -eq 0 ]
+	done
```
Comment (mrjana) [inline test/integration/dnet/multi.bats]: This is not correct. The intention of this test is to create a network from one node and check if that network can be seen from all the nodes and then try to delete the network from the 2nd node always (irrespective of where it was created)

Comment (rmb938) [inline test/integration/dnet/multi.bats]: Ah right that makes more sense. I'll have to do some debugging then as this test kept failing for some reason.

Comment (rmb938) [inline test/integration/dnet/multi.bats]: Hmm the tests worked fine this time. Odd

**Comment by aboch** [comment]
LGTM

**Comment by mavenugo** [comment]
@rmb938 Thanks for your patience. LGTM.

We are planning on a libnetwork vendor-in today into docker/docker.
This changes will be part of that. Please wait for it and complete your integration on the docker side.

**Pull request merged**